iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser.cpp

```diff
@@ -202,44 +206,44 @@ TEST_F(CmdLineParser_test, MonitoringModeOptionsLeadToCorrectMode)
         for (auto expectedValue : modeArray)
         {
             args[2] = valueArray[i];
-            testMonitoringMode(numberOfArgs, args, expectedValue);
+            testMonitoringMode(NUMBER_OF_ARGS, args, expectedValue);
             i++;
         }
     }
 }
 
 TEST_F(CmdLineParser_test, WrongMonitoringModeOptionLeadsToProgrammNotRunning)
 {
-    constexpr uint8_t numberOfArgs{3U};
-    char* args[numberOfArgs];
+    constexpr uint8_t NUMBER_OF_ARGS{3U};
+    char* args[NUMBER_OF_ARGS];
     char appName[] = "./foo";
     char option[] = "-m";
     char wrongValue[] = "DontBlink";
     args[0] = &appName[0];
     args[1] = &option[0];
     args[2] = &wrongValue[0];
 
-    iox::config::CmdLineParser sut;
-    auto result = sut.parse(numberOfArgs, args);
+    CmdLineParser sut;
+    auto result = sut.parse(NUMBER_OF_ARGS, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().run, Eq(false));
+    ASSERT_FALSE(result.has_error());
+    EXPECT_FALSE(result.value().run);
 }
 
 TEST_F(CmdLineParser_test, LogLevelOptionsLeadToCorrectLogLevel)
 {
-    constexpr uint8_t numberOfArgs{3U};
+    constexpr uint8_t NUMBER_OF_ARGS{3U};
     LogLevel loglevelArray[] = {LogLevel::kOff,
                                 LogLevel::kFatal,
                                 LogLevel::kError,
                                 LogLevel::kWarn,
                                 LogLevel::kInfo,
                                 LogLevel::kDebug,
                                 LogLevel::kVerbose};
-    char* args[numberOfArgs];
+    char* args[NUMBER_OF_ARGS];
     char appName[] = "./foo";
     char optionArray[][20] = {"-l", "--log-level"};
-    char valueArray[][10] = {"off", "fatal", "error", "warn", "info", "debug", "verbose"};
+    char valueArray[][10] = {"off", "fatal", "error", "warning", "info", "debug", "verbose"};
```

Comment:
I was wondering why ``warn`` worked before, so I tested an arbitrary letter sequence instead of ``warning`` which also works. @mossmaurice since you introduced the tests, do you have any idea what's wrong here?

Reply:
I can confirm that behaviour. Any letter sequence can be used instead of `warning`. The reason is that the default level is also `warning`. However, when taking the default branch in the switch case `iceoryx_posh/source/roudi/roudi_cmd_line_parser.cpp:146` `m_run` is set to false. Added an additional check for that.

Reply:
Once `m_run` is removed, we can use `INFO_OUTPUT_ONLY` for that purpose.

Reply:
So, is there an issue tracking `m_run` removal? Then we can mention it and close this conversation and PR.

Reply:
Yes, that's tracked in #539